R/missingness.R

```diff
@@ -134,3 +134,60 @@ countMissingData <- function(x, userNAs = NULL) {
   .Deprecated("missingness", "healthcareai")
   missingness(x)
 }
+
+#' Summarizes data given by \code{\link{missingness}}
+#'
+#' @description Interpreting \code{\link{missingness}} results from wide
+#'   datasets is difficult. This function helps interpret missingness output by
+#'   summarizing this output by listing: the percent of variables that
+#'   contain missingness, the variable name of the variable with the maximum
+#'   amount of missingness along with its percent missingness, and a tibble
+#'   that lists the top 5 missingness levels with the count of the number of
+#'   variables associated with each level.
+#' @param object Data frame from \code{\link{missingness}}
+#' @param ... Unused
+#' @return a tibble of the top 5 missingness percentage levels with the count of
+#'   the number of variables associated with each level
+#' @export
+#' @examples
+#' missingness(pima_diabetes) %>%
+#'   summary()
+#'
+summary.missingness <- function(object, ...) {
+  if (!length(object))
+    stop("`object` is empty.")
+
+  col_missing <- pull(object, percent_missing) > 0
+  if (sum(col_missing) == 0) {
+    stop("`object` has no variables with missingness.")
+  } else {
+    # Get the percent of columns that have any missing values
+    perc_col_missing <- mean(col_missing) * 100 # Convert from decimal to percent
+
+    # Get the name and percent_missing of the variable with the most missingness
+    max_df <- (
+      object %>% filter(percent_missing == max(percent_missing))
```

Comment:
`dplyr::top_n` is safer than this

Reply:
@michaellevy `dplyr::top_n` grabs more than one if there is a tie.  I don't think this is what we want. Maybe I am misunderstanding something. Is there something you want to use instead of returning to what I had before?

Reply:
I like `top_n` on line 189.

Reply:
Ah, good point. What you have might be fine, but in general creating objects that we don't know the dimensions of and then pulling the first row or whatever leaves more room for bugs to creep in than creating an object whose dimensions will always be the same. `which.max` is designed for what you're doing here and always returns a single index, so how about `object %>% dplyr::slice(which.max(percent_missing))`? (`slice` is the dplyr version of `d[i, ]` and in general tidyverse functions are safer than base-R functions.

Reply:
I wasn't aware of that function. Thank you.